src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -0,0 +1,478 @@
+
+        //Initialize the request
+        http_request refreshRequest(L"POST");
+        refreshRequest.set_request_uri(_tenantID + L"/oauth2/token");
+        utility::string_t body = L"client_id=" + AzureClientID + L"&resource=" + _wantedResource + L"&grant_type=refresh_token" + L"&refresh_token=" + _refreshToken;
+        refreshRequest.set_body(body, L"application/x-www-form-urlencoded");
+        refreshRequest.headers().add(L"User-Agent", L"Terminal");
+
+        //Send the request and return the response as a json value
+        return _requestHelper(refreshClient, refreshRequest);
+    }
+
+    // Helper function to get the user's cloud shell settings
+    json::value AzureConnection::_getCloudShellUserSettings()
+    {
+        //Initialize client
+        http_client settingsClient(_resourceUri);
+
+        //Initialize request
+        http_request settingsRequest(L"GET");
+        settingsRequest.set_request_uri(L"providers/Microsoft.Portal/userSettings/cloudconsole?api-version=2018-10-01");
+        settingsRequest.headers().add(L"Accept", L"application/json");
+        settingsRequest.headers().add(L"Content-Type", L"application/json");
+        settingsRequest.headers().add(L"Authorization", L"Bearer " + _accessToken);
+        settingsRequest.headers().add(L"User-Agent", L"Terminal");
+
+        return _requestHelper(settingsClient, settingsRequest);
+    }
+
+    utility::string_t AzureConnection::_getCloudShell()
+    {
+        //Initialize client
+        http_client cloudShellClient(_resourceUri);
+
+        //Initialize request
+        http_request shellRequest(L"PUT");
+        shellRequest.set_request_uri(L"providers/Microsoft.Portal/consoles/default?api-version=2018-10-01");
+        shellRequest.headers().add(L"Accept", L"application/json");
+        shellRequest.headers().add(L"Content-Type", L"application/json");
+        shellRequest.headers().add(L"Authorization", L"Bearer " + _accessToken);
+        shellRequest.headers().add(L"User-Agent", L"Terminal");
+        json::value innerBody = json::value::parse(U("{ \"osType\" : \"linux\" }"));
+        json::value body;
+        body[U("properties")] = innerBody;
+        shellRequest.set_body(body);
+
+        //Send the request and get the response as a json value
+        json::value cloudShell = _requestHelper(cloudShellClient, shellRequest);
+
+        //Return the uri
+        return cloudShell.at(L"properties").at(L"uri").as_string() + L"/";
+    }
+
+    utility::string_t AzureConnection::_getTerminal(utility::string_t shellType)
+    {
+        //Initialize client
+        http_client terminalClient(_cloudShellUri);
+
+        //Initialize the request
+        http_request terminalRequest(L"POST");
+        terminalRequest.set_request_uri(L"terminals?cols=" + std::to_wstring(_initialCols) + L"&rows=" + std::to_wstring(_initialRows) + L"&version=2019-01-01&shell=" + shellType);
+        terminalRequest.headers().add(L"Accept", L"application/json");

```

Comment:
I feel like you could write a helper method for `http_request` that applies the 4-6 common lines of headers to every request.